src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -228,6 +173,32 @@ public new CSharpParseOptions WithFeatures(IEnumerable<KeyValuePair<string, stri
             }
         }
 
+        internal override void ValidateOptions(ArrayBuilder<Diagnostic> builder)
+        {
+            ValidateOptions(builder, MessageProvider.Instance);
+
+            // Validate LanguageVersion not SpecifiedLanguageVersion, after Latest/Default has been converted:
+            if (!LanguageVersion.IsValid())
```

Comment:
I don't know why we're adding a diagnostic for this. `LanguageVersion` is something that is specified as an `enum`, not as a string that is parsed, which under our guidelines suggests that it should throw an ArgumentException when a bad language version is specified. The underlying bug to be fixed here doesn't suggest a diagnsotic for this.